### ISSUE TITLE
Support for Pydantic Version 2.0

### ISSUE THREAD

**Issue by yuvanist** [supabase-community/gotrue-py#282]
Currently `gotrue 1.0.2` depends on `pydantic<2.0.0 and >=1.10.0`. 
This creates friction for people who are migrating to Pydantic V2 as it is conflicting the requirements. Will be great if `gotrue-py` also does the transition to pydantic 2.0.

**Comment by titanism2197** [comment]
YES Indeed. i also has problem with `pydantic 2.0` and `gotrue 1.0.2`

here's the error i've got.
```
File "/Users/titanism/opt/miniconda3/envs/lastoria/lib/python3.10/site-packages/gotrue/types.py", line 80, in <module>
    class Session(BaseModel):
  File "/Users/titanism/opt/miniconda3/envs/lastoria/lib/python3.10/site-packages/gotrue/types.py", line 110, in Session
    def validator(cls, values: dict) -> dict:
  File "/Users/titanism/opt/miniconda3/envs/lastoria/lib/python3.10/site-packages/pydantic/deprecated/class_validators.py", line 222, in root_validator
    return root_validator()(*__args)  # type: ignore
  File "/Users/titanism/opt/miniconda3/envs/lastoria/lib/python3.10/site-packages/pydantic/deprecated/class_validators.py", line 228, in root_validator
    raise PydanticUserError(
pydantic.errors.PydanticUserError: If you use `@root_validator` with pre=False (the default) you MUST specify `skip_on_failure=True`. Note that `@root_validator` is deprecated and should be replaced with `@model_validator`.

For further information visit https://errors.pydantic.dev/2.1.1/u/root-validator-pre-skip
```

**Comment by yuvanist** [comment]
This is fixed in the release v1.0.3. 🎉
 https://github.com/supabase-community/gotrue-py/releases/tag/v1.0.3
